hydrate-orchestrator/utils.go

```diff
@@ -148,19 +151,58 @@ func (kd *KubernetesAppDeployment) Equals(other KubernetesAppDeployment) bool {
 		kd.Environment == other.Environment
 }
 
-func (kd *KubernetesAppDeployment) String(summary bool) string {
+func (kd *KubernetesAppDeployment) String(summary bool, repoURL ...string) string {
+	// Extract the necessary fields
+	serviceNames := kd.ServiceNames
+	repo := kd.RepositoryCaller
+	image := kd.Image
 
-	if summary {
+	// Build the repository link if a URL is provided
+	repoLink := repo
+	if len(repoURL) > 0 && repoURL[0] != "" {
+		repoLink = fmt.Sprintf("[%s](%s)", repo, repoURL[0])
+	}
 
-		return fmt.Sprintf(
-			"Deployment in cluster: `%s`, tenant: `%s`, env: `%s`",
-			kd.Cluster, kd.Tenant, kd.Environment,
-		)
+	if summary {
+		// If no service names are provided, just return the deployment coordinates
+		if len(serviceNames) == 0 {
+			return fmt.Sprintf(
+				"Deployment in cluster: `%s`, tenant: `%s`, env: `%s`",
+				kd.Cluster, kd.Tenant, kd.Environment,
+			)
+		} else {
+			// Use the first service_name or concatenate if multiple
+			serviceName := serviceNames[0]
+			if len(serviceNames) > 1 {
+				serviceName = strings.Join(serviceNames, ",")
+			}
+
+			return fmt.Sprintf(
+				"Deployment of `%s` in cluster: `%s`, tenant: `%s`, env: `%s`",
+				serviceName, kd.Cluster, kd.Tenant, kd.Environment,
+			)
+		}
 	} else {
-		return "Deployment coordinates:" +
-			fmt.Sprintf("\n\t* Cluster: `%s`", kd.Cluster) +
-			fmt.Sprintf("\n\t* Tenant: `%s`", kd.Tenant) +
-			fmt.Sprintf("\n\t* Environment: `%s`", kd.Environment)
+		// If no service names, repo link and image are provided, just return the deployment coordinates
+		if len(serviceNames) == 0 && repoLink == "" && image == "" {
```

Comment:
```suggestion
		if len(serviceNames) == 0 {
```
Too much conditions IMO, are all of them needed?